internal/nginx/config/servers_template.go

```diff
@@ -2,6 +2,15 @@ package config
 
 var serversTemplateText = `
 {{ range $s := . -}}
+
+    {{ range $m := $s.Maps }}
+map {{ $m.Source }} {{ $m.Variable }} {
+    {{ range $p := $m.Parameters }}
+    {{ $p.Value }} {{ $p.Result }};
+    {{ end }}
+}
```

Comment:
Prepend rather than append, but yes, exactly! The values will also be separated by a comma, as per [the spec](https://gateway-api.sigs.k8s.io/references/spec/#gateway.networking.k8s.io%2fv1beta1.HTTPHeaderFilter ).

Reply:
```
map ${http_accept_encoding} $accept_encoding {
    default '';
    ~^(?!\s*$).+ ${http_accept_encoding},;
}
map ${http_my_cool_header} $my_cool_header {
    default '';
    ~^(?!\s*$).+ ${http_my_cool_header},;
}
server {
    server_name cafe.example.com;
    location /coffee {
        proxy_set_header Accept-Encoding "${accept_encoding}compress";
        proxy_set_header My-cool-header "${my_cool_header}this-is-an-appended-value";
        proxy_set_header My-Overwrite-Header "this-is-the-only-value";
        proxy_set_header User-Agent "";
        proxy_set_header Host "$host";
        proxy_pass http://default_coffee_80$request_uri;
    }
    location / {
            return 404 "";
    }
}
```

Reply:
the approach with map doesn't work if the client send duplicate values of headers. For example:

```
curl localhost:9001/three -H "a: 1" -H "a: 2"
```

Response with we don't append header a:
```
Method: GET
HTTP version: 1.0
Host: test
Remote Address: 127.0.0.1
URI: /three
Headers:
  header 'Host' is 'test'
  header 'Connection' is 'close'
  header 'User-Agent' is 'curl/7.47.0'
  header 'Accept' is '*/*'
  header 'a' is '1,2'
Args:
```

Response if we append to the header  value '3' using the map approach:

```
Method: GET
HTTP version: 1.0
Host: test
Remote Address: 127.0.0.1
URI: /one
Headers:
  header 'a' is '1,3'
  header 'Host' is 'test'
  header 'Connection' is 'close'
  header 'User-Agent' is 'curl/7.47.0'
  header 'Accept' is '*/*'
Args:
```

Question to everyone: Shall we try to cover that case?

We can try to use njs. When we do header matching, we are able to match against all values of the header -- https://github.com/nginxinc/nginx-kubernetes-gateway/blob/406b8c67a5eebe8224f17b1b330e5ffa8f2fbfb7/internal/nginx/modules/src/httpmatches.js#L140-L153

Reply:
The spacing looks a little off when I curl with multiple header values and specify an "add" header modifier:

```
curl --resolve [cafe.example.com](http://cafe.example.com/):$GW_PORT:$GW_IP http://cafe.example.com/:$GW_PORT/coffee -H "a: 1" -H "a: 2"
```

with:
```
requestHeaderModifier:
  add:
  - name:  a
    value: "3"
```
I get:
```
  "requestHeaders": {
    "A": "1, 2,3",
  },
```
I think we need to add a space after the client headers.